src/js/backup_restore.js

```diff
@@ -190,7 +195,12 @@ function configuration_backup(callback) {
         MSP.promise(MSPCodes.MSP_ADVANCED_CONFIG).then(function() {
             return MSP.promise(MSPCodes.MSP_SENSOR_CONFIG);
         }).then(function() {
-            return MSP.promise(MSPCodes.MSP_NAME);
+            return semver.gte(FC.CONFIG.apiVersion, API_VERSION_1_45)
+                ? MSP.promise(MSPCodes.MSP2_GET_TEXT, mspHelper.crunch(MSPCodes.MSP2_GET_TEXT, MSPCodes.MSP2TEXT_CRAFT_NAME))
+                : MSP.promise(MSPCodes.MSP_NAME);
+        }).then(function() {
+            return semver.gte(FC.CONFIG.apiVersion, API_VERSION_1_45)
+                ? MSP.promise(MSPCodes.MSP2_GET_TEXT, mspHelper.crunch(MSPCodes.MSP2_GET_TEXT, MSPCodes.MSP2TEXT_PILOT_NAME)) : Promise.resolve(true);
```

Comment:
Anything returned from `Promise` is automatically wrapped in promise. Hence `resolve` is redundant.
```suggestion
            return semver.gte(FC.CONFIG.apiVersion, API_VERSION_1_45)
                ? MSP.promise(MSPCodes.MSP2_GET_TEXT, mspHelper.crunch(MSPCodes.MSP2_GET_TEXT, MSPCodes.MSP2TEXT_PILOT_NAME)) : true;
```